var/spack/repos/builtin/packages/care/package.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Care(CMakePackage, CudaPackage, HipPackage):
+    """
+    Algorithms for chai managed arrays.
+    """
+
+    homepage = "https://github.com/LLNL/CARE"
+    git      = "https://github.com/LLNL/CARE.git"
+
+    version('develop', branch='develop', submodules='True')
+    version('master', branch='main', submodules='True')
+    version('0.3.0', tag='v0.3.0', submodules='True')
+    version('0.2.0', tag='v0.2.0', submodules='True')
+
+    variant('openmp', default=False, description='Build Shared Libs')
+    variant('implicit_conversions', default=True, description='Enable implicit'
+            'conversions to/from raw pointers')
+    variant('benchmarks', default=True, description='Build benchmarks.')
+    variant('examples', default=True, description='Build examples.')
+    variant('docs', default=False, description='Build documentation')
+
+    depends_on('blt', type='build')
+
+    depends_on('camp')
+    depends_on('umpire@develop')
+    depends_on('raja@develop')
+    depends_on('chai@develop+enable_pick~benchmarks')
+
+    depends_on('camp+cuda', when='+cuda')
+    depends_on('umpire@develop+cuda', when='+cuda')
+    depends_on('raja@develop+cuda', when='+cuda')
+    depends_on('chai@develop+cuda+enable_pick~benchmarks', when='+cuda')
```

Comment:
You don't need to merge the unconditional portions of the constraints with the conditional portions here -- the unconditional portions are already defined above.

```suggestion
    depends_on('umpire+cuda', when='+cuda')
    depends_on('raja+cuda', when='+cuda')
    depends_on('chai+cuda', when='+cuda')
```

Reply:
Will test after lunch - need to address this still

Reply:
Just did this, off to lunch now. All comments addressed now!